ehrql/loaders.py

```diff
@@ -59,6 +88,70 @@ def load_definition_in_subprocess(definition_type, definition_file, user_args, e
     return deserialize(result.stdout)
 
 
+def isolation_is_supported():
+    return subprocess.run([PLEDGE_BIN, "-T", "pledge"]).returncode == 0
```

Comment:
This raises an `OSError` on macOS. For example:

```
OSError: [Errno 8] Exec format error: PosixPath('/Users/iaindillingham/Code/opensafely-core/ehrql/bin/pledge')
```

Would the following be reasonable?

```suggestion
    try:
        return subprocess.run([PLEDGE_BIN, "-T", "pledge"]).returncode == 0
    except OSError:
        return False
```